inference-engine/src/vpu/myriad_plugin/myriad_plugin.cpp

```diff
@@ -98,24 +135,128 @@ void Engine::QueryNetwork(
         VPU_THROW_UNLESS(!(std::find(deviceIDs.begin(), deviceIDs.end(), deviceName) == deviceIDs.end()), "Myriad device: {} not found.", deviceName);
     }
 
-    if (network.getFunction()) {
-        THROW_IE_EXCEPTION << NOT_IMPLEMENTED_str << " ngraph::Function is not supported natively";
-    }
+    auto function = network.getFunction();
+
+    if (function != nullptr) {
+        std::unordered_set<std::string> originalOps;
+        for (auto&& node : function->get_ops()) {
+            if (!ngraph::op::is_parameter(node) &&
+                !ngraph::op::is_output(node) &&
+                !ngraph::op::is_constant(node)) {
+                originalOps.emplace(node->get_friendly_name());
+            }
+        }
 
-    const auto log = std::make_shared<Logger>(
-        "GraphCompiler",
-        parsedConfigCopy.logLevel(),
-        defaultOutput(parsedConfigCopy.compilerLogFilePath()));
+        auto clonedNetwork = CloneAndTransformNetwork(network);
+
+        std::unordered_set<std::string> supported;
+        std::unordered_set<std::string> unsupported;
+
+        std::unordered_set<std::string> splitNames;
+        std::unordered_set<std::string> concatNames;
+
+        std::vector<std::shared_ptr<ngraph::Node>> splits;
+        std::vector<std::shared_ptr<ngraph::Node>> concats;
+
+        for (CNNNetworkIterator itLayer{clonedNetwork.get()};
+             itLayer != CNNNetworkIterator();
+             itLayer++) {
+            auto layerIsSupported = [&] {
+                auto node = (*itLayer)->getNode();
+
+                if (std::dynamic_pointer_cast<const ::ngraph::opset3::Split>(node) != nullptr) {
+                    splitNames.emplace(node->get_friendly_name());
+                    splits.push_back(node);
+                    return false;
```

Comment:
It is not clear for me why Split and Concat are not supported. Could you explain, please?
Add the corresponding comments as well

Reply:
We marked Concat and Split as unsupported because we will check them later: if all parent layers is supported, only in this case we will mark Concat as a supported for VPU, similarly for Split.